notification-config-service-impl/src/main/java/org/hypertrace/notification/config/service/NotificationConfigServiceRequestValidator.java

```diff
@@ -0,0 +1,111 @@
+package org.hypertrace.notification.config.service;
+
+import static org.hypertrace.config.validation.GrpcValidatorUtils.validateRequestContextOrThrow;
+
+import com.google.common.base.Preconditions;
+import org.hypertrace.core.grpcutils.context.RequestContext;
+import org.hypertrace.notification.config.service.v1.CreateNotificationChannelRequest;
+import org.hypertrace.notification.config.service.v1.CreateNotificationRuleRequest;
+import org.hypertrace.notification.config.service.v1.DeleteNotificationChannelRequest;
+import org.hypertrace.notification.config.service.v1.DeleteNotificationRuleRequest;
+import org.hypertrace.notification.config.service.v1.GetAllNotificationChannelsRequest;
+import org.hypertrace.notification.config.service.v1.GetAllNotificationRulesRequest;
+import org.hypertrace.notification.config.service.v1.UpdateNotificationChannelRequest;
+import org.hypertrace.notification.config.service.v1.UpdateNotificationRuleRequest;
+
+public class NotificationConfigServiceRequestValidator {
+
+  public void validateCreateNotificationRuleRequest(
+      RequestContext requestContext, CreateNotificationRuleRequest request) {
+    validateRequestContextOrThrow(requestContext);
+    Preconditions.checkArgument(
```

Comment:
These should switch to use common validator utils (less code and better error messaging).

Reply:
But the utils can't be used to check the presence of message type of field?

Reply:
This is by design - checking `hasNewNotificationRule` here is redundant with the later checks checking that values inside are valid. We validate on primitives (that is, leaf nodes of the message), so if we don't require any of the primitives inside a message, there's no reason to require the message.

Reply:
yeah makes sense, if primitive is not required then the message is not required.